src/main/java/bsh/Operators.java

```diff
@@ -399,9 +401,62 @@ static Object bigIntegerBinaryOperation(BigInteger lhs, BigInteger rhs, int kind
                 "Unimplemented binary integer operator");
     }
 
+    // returns Object covering both Float return types
+    static Object floatBinaryOperation(float lhs, float rhs, int kind)
+            throws UtilEvalError
+    {
+        switch(kind)
+        {
+            // arithmetic
+            case PLUS:
+                if ( lhs > 0d && (Float.MAX_VALUE - lhs) < rhs )
+                    break;
+                return lhs + rhs;
+
+            case MINUS:
+                if ( lhs < 0d && (-Float.MAX_VALUE - lhs) > -rhs )
+                    break;
+                return lhs - rhs;
+
+            case STAR:
+                if ( lhs != 0 && Float.MAX_VALUE / lhs < rhs )
+                    break;
+                return lhs * rhs;
+
+            case SLASH:
+                return lhs / rhs;
+
+            case MOD:
+            case MODX:
+                return lhs % rhs;
+
+            case POWER:
+            case POWERX:
+                double check = Math.pow(lhs, rhs);
+                if ( Double.isInfinite(check) )
```

Comment:
Should we not try and return a float here?

Reply:
Math.pow returns a double, so no one is expecting to get a float here.